dlt-connector/src/data/proto/3_3/TransactionBody.ts

```diff
@@ -0,0 +1,129 @@
+        type = determineCrossGroupType(transaction)
+        otherGroup = determineOtherGroup(type, transaction)
+      }
+
+      super({
+        memo: 'Not implemented yet',
+        createdAt: new Timestamp(new Date(transaction.createdAt)),
+        versionNumber: '3.3',
+        type,
+        otherGroup,
+      })
+    } else {
+      super()
+    }
+  }
+
+  @Field.d(1, 'string')
+  public memo: string
+
+  @Field.d(2, Timestamp)
+  public createdAt: Timestamp
+
+  @Field.d(3, 'string')
+  public versionNumber: string
+
+  @Field.d(4, CrossGroupType)
+  public type: CrossGroupType
+
+  @Field.d(5, 'string')
+  public otherGroup: string
+
+  @OneOf.d(
+    'gradidoTransfer',
+    'gradidoCreation',
+    'groupFriendsUpdate',
+    'registerAddress',
+    'gradidoDeferredTransfer',
+    'communityRoot',
+  )
+  public data: string
+
+  @Field.d(6, 'GradidoTransfer')
+  transfer?: GradidoTransfer
+
+  @Field.d(7, 'GradidoCreation')
+  creation?: GradidoCreation
+
+  @Field.d(8, 'GroupFriendsUpdate')
+  groupFriendsUpdate?: GroupFriendsUpdate
+
+  @Field.d(9, 'RegisterAddress')
+  registerAddress?: RegisterAddress
+
+  @Field.d(10, 'GradidoDeferredTransfer')
+  deferredTransfer?: GradidoDeferredTransfer
+
+  @Field.d(11, 'CommunityRoot')
+  communityRoot?: CommunityRoot
+
+  public getTransactionType(): TransactionType | undefined {
+    if (this.transfer) return TransactionType.GRADIDO_TRANSFER
+    else if (this.creation) return TransactionType.GRADIDO_CREATION
+    else if (this.groupFriendsUpdate) return TransactionType.GROUP_FRIENDS_UPDATE
+    else if (this.registerAddress) return TransactionType.REGISTER_ADDRESS
+    else if (this.deferredTransfer) return TransactionType.GRADIDO_DEFERRED_TRANSFER
+    else if (this.communityRoot) return TransactionType.COMMUNITY_ROOT
+  }
+
+  public getTransactionBase(): TransactionBase | undefined {
+    if (this.transfer) return this.transfer
+    if (this.creation) return this.creation
+    if (this.groupFriendsUpdate) return this.groupFriendsUpdate
+    if (this.registerAddress) return this.registerAddress
+    if (this.deferredTransfer) return this.deferredTransfer
+    if (this.communityRoot) return this.communityRoot
+  }
+
+  public fillTransactionRecipe(recipe: Transaction): void {
+    recipe.createdAt = timestampToDate(this.createdAt)
+    recipe.protocolVersion = this.versionNumber
+    const transactionType = this.getTransactionType()
+    if (!transactionType) {
+      throw new LogError("invalid TransactionBody couldn't determine transaction type")
+    }
+    recipe.type = transactionType.valueOf()
+    this.getTransactionBase()?.fillTransactionRecipe(recipe)
+  }
+
+  public getRecipientPublicKey(): Buffer | undefined {
+    if (this.transfer) {
+      return this.transfer.recipient
+    }
+    if (this.creation) {
+      return this.creation.recipient.pubkey
+    }
+    if (this.deferredTransfer) {
+      return this.deferredTransfer.transfer.recipient
+    }
```

Comment:
This is the protobuf basic definition for gradido on which this is based: 
https://github.com/gradido/gradido_protocol/blob/master/proto/gradido/gradido_transfer.proto
I have always tried to using short names for it. 
Also the C++ Code is based on it. Renaming make a lot of work. 
I could only rename it in the javascript implementation, because protobuf use intern the numbers not the names for serialization so it stay compatible. But it can lead to confusion later if others inspect the whole code. 
For me it was always clear that it was a public key because it was Type bytes. 
How important is it to you that I change it?

Reply:
Ok, I have added a comment for this.